src/macros.jl

```diff
@@ -543,6 +556,18 @@ function build_constraint(
     return VectorConstraint(x, MOI.SOS2{Float64}(set.weights))
 end
 
+function build_non_relational_constraint(_error::Function, head, args...) # This would remain in JuMP.
+    _error("Constraint $(head) is not recognised.")
+end
+
+function build_non_relational_constraint(_error::Function, head::Val{:alldifferent}, args::VariableRef...) # This would be an extension to JuMP.
```

Comment:
We could redirect `isequal` and `isless` to the build call with `EqualTo`/`LessThan`/`GreaterThan`